### PR TITLE
feat:  Add esbuild-plugins package

### PR DESCRIPTION
**Pull request by GordonSmith**
## Checklist:
- [ ] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

File: packages/esbuild-plugins/src/build.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        describe: "Watch for changes",
+        type: "boolean"
+    })
+    .help("h")
+    .alias("h", "help")
+    .epilog("https://github.com/hpcc-systems/hpcc-js-wasm")
+    ;
+
+export const pkg = JSON.parse(readFileSync(path.join(process.cwd(), "./package.json"), "utf8"));
+export const NODE_MJS = pkg.type === "module" ? "js" : "mjs";
```
Comment (jeclrsg) [inline packages/esbuild-plugins/src/build.ts:34]: Are these backwards, if type === module then mjs?

File: packages/esbuild-plugins/src/sfx-wrapper.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    }
+} `.trim();
+}
+
+export async function wrap(path: string) {
+    const base91 = await Base91.load();
+    const zstd = await Zstd.load();
+
+    const wasm = await readFile(path);
+    path = path.replace(/\.js$/, ".xxx");
```
Comment (jeclrsg) [inline packages/esbuild-plugins/src/sfx-wrapper.ts:85]: Is this replacement correct? It didn't seem to get converted back anywhere.

**Review by jeclrsg — COMMENTED** [review]
@GordonSmith just a couple of questions

File: packages/esbuild-plugins/src/build.ts (hunk 1 above)
Comment (GordonSmith) [reply packages/esbuild-plugins/src/build.ts:34]: No - if the type is "module" then it expects all "js" files to be modern "import" style.

File: packages/esbuild-plugins/src/sfx-wrapper.ts (hunk 2 above)
Comment (GordonSmith) [reply packages/esbuild-plugins/src/sfx-wrapper.ts:85]: Its a bit circular, but without this you would end up with an infinite loop...

**Review by jeclrsg — APPROVED** [review]

**Pull request merged**